File: 3 Time Complexity/TapeEquilibrium/tapeEquilibrium.java
class Solution {
  public int solution(int[] A) {
    int totalSum = 0;
    int min = Integer.MAX_VALUE;

    for (int i : A) totalSum += i;

    //System.out.println("total sum: " + totalSum);
    for (int i = 0; i < A.length - 1; i++) {
      totalSum -= 2 * A[i];
      if (min > Math.abs(totalSum)) {
        min = Math.abs(totalSum);
      }
    }
    //System.out.println("min " + min);
    return min;
  }
}

File: 3 Time Complexity/TapeEquilibrium/tapeEquilibrium.cpp
#include <limits.h>
#include <cmath>

int solution(vector<int> &A) {
  int totalSum = 0;
  int min = INT_MAX;

  for (int i : A) totalSum += i;
  for (unsigned int i = 0; i < A.size() - 1; i++) {
    totalSum -= 2 * A[i];
    if (min > std::abs(totalSum)) {
      min = std::abs(totalSum);
    }
  }
  return min;
}